### PR TITLE
ui: fix combine endpoint when activity table is empty

### LINKED ISSUE #111320: sql activity failed to load
1. Start demo cluster
2. Run for a while
3. Stop running and leave it idle (so there is no recent data)
4. Try to retrieve the last hour

The page will fail to load, and the error on logs show:

```
E230927 00:44:36.381660 257330 server/combined_statement_stats.go:172  [T1,Vsystem,n1,status] 4555  expected one row but got none on getSourceStatsInfo.getRuntime
E230927 00:44:36.381660 257330 server/combined_statement_stats.go:172  [T1,Vsystem,n1,status] 4555 +(1) attached stack trace
E230927 00:44:36.381660 257330 server/combined_statement_stats.go:172  [T1,Vsystem,n1,status] 4555 +  -- stack trace:
E230927 00:44:36.381660 257330 server/combined_statement_stats.go:172  [T1,Vsystem,n1,status] 4555 +  | github.com/cockroachdb/cockroach/pkg/server.getSourceStatsInfo.func1
E230927 00:44:36.381660 257330 server/combined_statement_stats.go:172  [T1,Vsystem,n1,status] 4555 +  | 	github.com/cockroachdb/cockroach/pkg/server/combined_statement_stats.go:341
E230927 00:44:36.381660 257330 server/combined_statement_stats.go:172  [T1,Vsystem,n1,status] 4555 +  | github.com/cockroachdb/cockroach/pkg/server.getSourceStatsInfo
E230927 00:44:36.381660 257330 server/combined_statement_stats.go:172  [T1,Vsystem,n1,status] 4555 +  | 	github.com/cockroachdb/cockroach/pkg/server/combined_statement_stats.go:397
E230927 00:44:36.381660 257330 server/combined_statement_stats.go:172  [T1,Vsystem,n1,status] 4555 +  | github.com/cockroachdb/cockroach/pkg/server.getCombinedStatementStats
E230927 00:44:36.381660 257330 server/combined_statement_stats.go:172  [T1,Vsystem,n1,status] 4555 +  | 	github.com/cockroachdb/cockroach/pkg/server/combined_statement_stats.go:162
E230927 00:44:36.381660 257330 server/combined_statement_stats.go:172  [T1,Vsystem,n1,status] 4555 +  | github.com/cockroachdb/cockroach/pkg/server.(*statusServer).CombinedStatementStats
E230927 00:44:36.381660 257330 server/combined_statement_stats.go:172  [T1,Vsystem,n1,status] 4555 +  | 	github.com/cockroachdb/cockroach/pkg/server/combined_statement_stats.go:67
E230927 00:44:36.381660 257330 server/combined_statement_stats.go:172  [T1,Vsystem,n1,status] 4555 +  | github.com/cockroachdb/cockroach/pkg/server/serverpb._Status_CombinedStatementStats_Handler.func1
E230927 00:44:36.381660 257330 server/combined_statement_stats.go:172  [T1,Vsystem,n1,status] 4555 +  | 	github.com/cockroachdb/cockroach/pkg/server/serverpb/bazel-out/darwin_arm64-fastbuild/bin/pkg/server/serverpb/serverpb_go_proto_/github.com/cockroachdb/cockroach/pkg/server/serverpb/status.pb.go:9755
E230927 00:44:36.381660 257330 server/combined_statement_stats.go:172  [T1,Vsystem,n1,status] 4555 +  | github.com/cockroachdb/cockroach/pkg/util/tracing/grpcinterceptor.ServerInterceptor.func1
E230927 00:44:36.381660 257330 server/combined_statement_stats.go:172  [T1,Vsystem,n1,status] 4555 +  | 	github.com/cockroachdb/cockroach/pkg/util/tracing/grpcinterceptor/grpc_interceptor.go:116
E230927 00:44:36.381660 257330 server/combined_statement_stats.go:172  [T1,Vsystem,n1,status] 4555 +  | google.golang.org/grpc.getChainUnaryHandler.func1
E230927 00:44:36.381660 257330 server/combined_statement_stats.go:172  [T1,Vsystem,n1,status] 4555 +  | 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1163
E230927 00:44:36.381660 257330 server/combined_statement_stats.go:172  [T1,Vsystem,n1,status] 4555 +  | github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func3
E230927 00:44:36.381660 257330 server/combined_statement_stats.go:172  [T1,Vsystem,n1,status] 4555 +  | 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:169
E230927 00:44:36.381660 257330 server/combined_statement_stats.go:172  [T1,Vsystem,n1,status] 4555 +  | google.golang.org/grpc.getChainUnaryHandler.func1
E230927 00:44:36.381660 257330 server/combined_statement_stats.go:172  [T1,Vsystem,n1,status] 4555 +  | 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1163
E230927 00:44:36.381660 257330 server/combined_statement_stats.go:172  [T1,Vsystem,n1,status] 4555 +  | github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func1.1
E230927 00:44:36.381660 257330 server/combined_statement_stats.go:172  [T1,Vsystem,n1,status] 4555 +  | 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:136
E230927 00:44:36.381660 257330 server/combined_statement_stats.go:172  [T1,Vsystem,n1,status] 4555 +  | github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTaskWithErr
E230927 00:44:36.381660 257330 server/combined_statement_stats.go:172  [T1,Vsystem,n1,status] 4555 +  | 	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:336
E230927 00:44:36.381660 257330 server/combined_statement_stats.go:172  [T1,Vsystem,n1,status] 4555 +  | github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func1
E230927 00:44:36.381660 257330 server/combined_statement_stats.go:172  [T1,Vsystem,n1,status] 4555 +  | 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:134
E230927 00:44:36.381660 257330 server/combined_statement_stats.go:172  [T1,Vsystem,n1,status] 4555 +  | google.golang.org/grpc.chainUnaryInterceptors.func1
E230927 00:44:36.381660 257330 server/combined_statement_stats.go:172  [T1,Vsystem,n1,status] 4555 +  | 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1154
E230927 00:44:36.381660 257330 server/combined_statement_stats.go:172  [T1,Vsystem,n1,status] 4555 +  | github.com/cockroachdb/cockroach/pkg/server/serverpb._Status_CombinedStatementStats_Handler
E230927 00:44:36.381660 257330 server/combined_statement_stats.go:172  [T1,Vsystem,n1,status] 4555 +  | 	github.com/cockroachdb/cockroach/pkg/server/serverpb/bazel-out/darwin_arm64-fastbuild/bin/pkg/server/serverpb/serverpb_go_proto_/github.com/cockroachdb/cockroach/pkg/server/serverpb/status.pb.go:9757
E230927 00:44:36.381660 257330 server/combined_statement_stats.go:172  [T1,Vsystem,n1,status] 4555 +  | google.golang.org/grpc.(*Server).processUnaryRPC
E230927 00:44:36.381660 257330 server/combined_statement_stats.go:172  [T1,Vsystem,n1,status] 4555 +  | 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1336
E230927 00:44:36.381660 257330 server/combined_statement_stats.go:172  [T1,Vsystem,n1,status] 4555 +  | google.golang.org/grpc.(*Server).handleStream
E230927 00:44:36.381660 257330 server/combined_statement_stats.go:172  [T1,Vsystem,n1,status] 4555 +  | 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1704
E230927 00:44:36.381660 257330 server/combined_statement_stats.go:172  [T1,Vsystem,n1,status] 4555 +  | google.golang.org/grpc.(*Server).serveStreams.func1.2
E230927 00:44:36.381660 257330 server/combined_statement_stats.go:172  [T1,Vsystem,n1,status] 4555 +  | 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:965
E230927 00:44:36.381660 257330 server/combined_statement_stats.go:172  [T1,Vsystem,n1,status] 4555 +  | runtime.goexit
E230927 00:44:36.381660 257330 server/combined_statement_stats.go:172  [T1,Vsystem,n1,status] 4555 +  | 	src/runtime/asm_arm64.s:1172
E230927 00:44:36.381660 257330 server/combined_statement_stats.go:172  [T1,Vsystem,n1,status] 4555 +Wraps: (2) expected one row but got none on getSourceStatsInfo.getRuntime
E230927 00:44:36.381660 257330 server/combined_statement_stats.go:172  [T1,Vsystem,n1,status] 4555 +Error types: (1) *withstack.withStack (2) *errutil.leafError
E230927 00:44:40.803654 420 1@server/status/runtime.go:501  [T1,Vsystem,n1] 4556  unable to get CPU capacity: failed to read cpu,cpuacct cgroup from cgroups file: /proc/self/cgroup: open /proc/self/cgroup: no such file or directory

Jira issue: CRDB-31855

### PR DESCRIPTION
**Pull request by j82w**
problem:
The combine endpoint would return an error if the activity table did not have any rows for the specificied time, and had rows for an older time frame. This endpoint returned correctly if the activity tables were completely empty.

Error:
expected one row but got none on getSourceStatsInfo.getRuntime attached stack trace
-- stack trace:
github.com/cockroachdb/cockroach/pkg/server.getSourceStatsInfo.func1
  github.com/cockroachdb/cockroach/pkg/server/combined_statement_stats.go:341
github.com/cockroachdb/cockroach/pkg/server.getSourceStatsInfo
  github.com/cockroachdb/cockroach/pkg/server/combined_statement_stats.go:397
github.com/cockroachdb/cockroach/pkg/server.getCombinedStatementStats
  github.com/cockroachdb/cockroach/pkg/server/combined_statement_stats.go:162
github.com/cockroachdb/cockroach/pkg/server.(*statusServer).CombinedStatementStats
  github.com/cockroachdb/cockroach/pkg/server/combined_statement_stats.go:67

Solution:
Return 0 instead of returning an error, and pass in the flag to the method to match the fallback logic.

Fixes: #111320

Release note (ui change): Fixes an error on the activity page when there was a workload and then the workload stopped so no queries ran against the db in the last hour.

### CODE REVIEW

**Review by maryliag — APPROVED** [review]
:lgtm:

> Reviewed 3 of 3 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/111420#-:-NfRoODe69dUMfuU23lz:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @ericharmeling)

File: pkg/server/combined_statement_stats.go [hunk 1]
```diff
@@ -304,9 +304,9 @@ func getSourceStatsInfo(
 	}
 	whereClauseOldestDate := buffer.String()
 
-	getRuntime := func(table string) (float32, error) {
+	getRuntime := func(table string, useActivityTable bool) (float32, error) {
 		var queryToGetClusterTotalRunTime string
-		if activityTableHasAllData {
+		if useActivityTable {
```
Comment (xinhaoz) [inline pkg/server/combined_statement_stats.go:309]: what if we just check if the table is the activity table?

**Comment by j82w** [comment]
*[`pkg/server/combined_statement_stats.go` line 309 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/111420#gh-1340509930:-NfS1fnL3k1k_q_m5cFb:betusmh) ([raw file](https://github.com/cockroachdb/cockroach/blob/b0f3d87288ee7d8bda989b67e5c69f9bb83d8481/pkg/server/combined_statement_stats.go#L309)):*
<details><summary><i>Previously, xinhaoz (Xin Hao Zhang) wrote…</i></summary><blockquote>

what if we just check if the table is the activity table?
</blockquote></details>

How about the new approach to pass in the func to create the query? Trying to avoid the overhead of doing string comparisons.

File: pkg/server/combined_statement_stats.go [hunk 2]
```diff
@@ -337,8 +321,10 @@ FROM %s %s`, table, whereClause)
 		if err != nil {
 			return 0, err
 		}
+
+		// It's possible the table is empty. Just return 0.
```
Comment (xinhaoz) [inline pkg/server/combined_statement_stats.go:325]: Unrelated, but should that defer func closing the iterator be moved up to after the first error check after the query?

**Review by xinhaoz — COMMENTED** [review]
LGTM, but I think for the commit msg `ui` should be `server` as this is in the server pkg.

**Review by xinhaoz — APPROVED** [review]

**Comment by j82w** [comment]
*[`pkg/server/combined_statement_stats.go` line 325 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/111420#gh-1340545019:-NfSQlzL4wcx4kOvr7kH:bpp89vb) ([raw file](https://github.com/cockroachdb/cockroach/blob/3b26273488638742e26094183df27708e670ed62/pkg/server/combined_statement_stats.go#L325)):*
<details><summary><i>Previously, xinhaoz (Xin Hao Zhang) wrote…</i></summary><blockquote>

Unrelated, but should that defer func closing the iterator be moved up to after the first error check after the query?
</blockquote></details>

Yes, it should call defer after the first error check. I went ahead and fixed it.

**Pull request merged**